File: iOS-v0/Sources/Models/ContentModels.swift
import Foundation

// MARK: - 首页信息流枚举（对齐 Android HomeFeedType / HomeFeedTargetType）
enum HomeFeedType: String, Codable, Sendable {
  case hotPost = "hot_post"
  case hotCompany = "hot_company"
  case hotJob = "hot_job"
}

enum HomeFeedTargetType: String, Codable, Sendable {
  case post
  case company
  case job
}

// MARK: - 首页混排卡片（严格对齐 Android HomeFeedItem）
struct HomeFeedItem: Codable, Identifiable, Sendable {
  let id: String
  let type: HomeFeedType
  let targetType: HomeFeedTargetType
  let targetId: String
  let title: String
  let summary: String?
  let imageUrl: String?
  let tags: [String]
  let authorName: String
  let authorAvatar: String?
  let badge: String
  let metricLabel: String?
  let metricValue: String?
  let createdAt: String?
}

// MARK: - 用户帖子
struct UserPost: Codable, Identifiable, Sendable {
  let id: String
  let title: String
  let content: String
  let coverImage: String?
  /// 服务端偶尔返回 null，统一走 optional
  let images: [String]?
  let tags: [String]
  let viewCount: Int
  let likeCount: Int
  let commentCount: Int
  let shareCount: Int?
  let createdAt: String
  let author: UserPostAuthor?
}

struct UserPostAuthor: Codable, Identifiable, Hashable, Sendable {
  let id: String?
  let name: String?
  let avatar: String?
  let headline: String?
}

// MARK: - 大咖分享
struct ExpertPost: Codable, Identifiable, Sendable {
  let id: String
  let expertName: String
  let expertTitle: String
  let expertCompany: String
  let expertAvatar: String?
  let title: String
  let content: String
  let coverImage: String?
  let tags: [String]
  let viewCount: Int
  let likeCount: Int
  let commentCount: Int?
  let publishedAt: String?
}

// MARK: - 推广职位
struct PromotedJob: Codable, Identifiable, Sendable {
  var id: String { promotionId }
  let promotionId: String
  let promotionType: String // NORMAL | PREMIUM | FEATURED
  let job: JobInfo
}

struct JobInfo: Codable, Identifiable, Sendable {
  let id: String
  let title: String
  let salary: String?
  let location: String?
  let skills: [String]
  let company: CompanyInfo
}

struct CompanyInfo: Codable, Identifiable, Sendable {
  let id: String
  let name: String
  let logo: String?
  let industry: String?
}

// MARK: - Banner
struct Banner: Codable, Identifiable, Sendable {
  let id: String
  let title: String
  let subtitle: String
  let description: String
  let imageUrl: String
  let linkType: String? // post | assessment | company
  let linkId: String?
}

// MARK: - 首页精选文章
struct HomeFeaturedArticle: Codable, Identifiable, Sendable {
  let id: String
  let title: String
  let summary: String?
  let imageUrl: String
  let author: String?
  let tags: [String]
  let viewCount: Int
  let category: String?
  let createdAt: String?
}

// MARK: - 互动状态
struct PostEngagement: Codable, Sendable {
  let postId: String
  let postType: String
  let likeCount: Int
  let commentCount: Int
  let favoriteCount: Int
  let isLiked: Bool
  let isFavorited: Bool
}

// MARK: - 评论
struct PostCommentAuthor: Codable, Identifiable, Hashable, Sendable {
  let id: String?
  let name: String?
  let avatar: String?
}

struct PostComment: Codable, Identifiable, Sendable {
  let id: String
  let content: String
  let createdAt: String
  let author: PostCommentAuthor
  let parentId: String?
  let replyToUserId: String?
  let replyToUserName: String?
  let likeCount: Int?
  let replyCount: Int?
  let isLiked: Bool?
  let reactions: [String: Int]?
  let replies: [PostComment]?
}

struct CreatePostCommentRequest: Encodable, Sendable {
  let content: String
  let parentId: String?
  let replyToUserId: String?
}

struct CreatePostCommentResult: Decodable, Sendable {
  let comment: PostComment?
  let engagement: PostEngagement?
}

struct CommentReactionRequest: Encodable, Sendable {
  let emoji: String
}

struct CommentReactionResult: Decodable, Sendable {
  let comment: PostComment?
  let added: Bool?
}

File: android-v0-compose/app/src/main/java/com/example/v0clone/data/model/ContentModels.kt
package com.xlwl.AiMian.data.model

import com.google.gson.annotations.SerializedName

/**
 * 内容类型
 */
enum class HomeFeedType {
    @SerializedName("hot_post")
    HOT_POST,

    @SerializedName("hot_company")
    HOT_COMPANY,

    @SerializedName("hot_job")
    HOT_JOB
}

enum class HomeFeedTargetType {
    @SerializedName("post")
    POST,

    @SerializedName("company")
    COMPANY,

    @SerializedName("job")
    JOB
}

/**
 * 首页内容卡片（混排）
 */
data class HomeFeedItem(
    val id: String,
    val type: HomeFeedType,
    val targetType: HomeFeedTargetType,
    val targetId: String,
    val title: String,
    val summary: String? = null,
    val imageUrl: String? = null,
    val tags: List<String> = emptyList(),
    val authorName: String,
    val authorAvatar: String? = null,
    val badge: String,
    val metricLabel: String? = null,
    val metricValue: String? = null,
    val createdAt: String? = null
)

/**
 * 用户帖子（热门分享）
 */
data class UserPost(
    val id: String,
    val title: String,
    val content: String,
    val coverImage: String?,
    val images: List<String>,
    val tags: List<String>,
    val viewCount: Int,
    val likeCount: Int,
    val commentCount: Int,
    val shareCount: Int = 0,
    val createdAt: String,
    val author: UserPostAuthor? = null
)

data class UserPostAuthor(
    val id: String?,
    val name: String?,
    val avatar: String?,
    val headline: String?
)

/**
 * 大咖分享
 */
data class ExpertPost(
    val id: String,
    val expertName: String,
    val expertTitle: String,
    val expertCompany: String,
    val expertAvatar: String?,
    val title: String,
    val content: String,
    val coverImage: String?,
    val tags: List<String>,
    val viewCount: Int,
    val likeCount: Int,
    val commentCount: Int = 0,
    val publishedAt: String? = null
)

/**
 * 推广职位
 */
data class PromotedJob(
    val promotionId: String,
    val promotionType: String, // NORMAL | PREMIUM | FEATURED
    val job: JobInfo
)

data class JobInfo(
    val id: String,
    val title: String,
    val salary: String?,
    val location: String?,
    val skills: List<String>,
    val company: CompanyInfo
)

data class CompanyInfo(
    val id: String,
    val name: String,
    val logo: String?,
    val industry: String?
)

/**
 * Banner 数据
 */
data class Banner(
    val id: String,
    val title: String,
    val subtitle: String,
    val description: String,
    val imageUrl: String,
    val linkType: String?, // post | assessment | company
    val linkId: String?
)

/**
 * 首页精选内容
 */
data class HomeFeaturedArticle(
    val id: String,
    val title: String,
    val summary: String?,
    val imageUrl: String,
    val author: String?,
    val tags: List<String> = emptyList(),
    val viewCount: Int = 0,
    val category: String?,
    val createdAt: String?
)

data class PostEngagement(
    val postId: String,
    val postType: String,
    val likeCount: Int,
    val commentCount: Int,
    val favoriteCount: Int,
    val isLiked: Boolean,
    val isFavorited: Boolean
)

data class PostCommentAuthor(
    val id: String?,
    val name: String?,
    val avatar: String?
)

data class PostCommentDto(
    val id: String,
    val content: String,
    val createdAt: String,
    val author: PostCommentAuthor,
    val parentId: String? = null,
    val replyToUserId: String? = null,
    val replyToUserName: String? = null,
    val likeCount: Int = 0,
    val replyCount: Int = 0,
    val isLiked: Boolean = false,
    val reactions: Map<String, Int> = emptyMap(),
    val replies: List<PostCommentDto> = emptyList()
)

data class CreatePostCommentRequest(
    val content: String,
    val parentId: String? = null,
    val replyToUserId: String? = null
)

data class CreatePostCommentResult(
    val comment: PostCommentDto?,
    val engagement: PostEngagement?
)

data class CommentReactionRequest(
    val emoji: String
)

data class CommentReactionResult(
    val comment: PostCommentDto?,
    val added: Boolean? = null
)
